psdk_wrapper/src/psdk_wrapper.cpp

```diff
@@ -117,6 +118,16 @@ PSDKWrapper::on_activate(const rclcpp_lifecycle::State &state)
   {
     publish_static_transforms();
   }
+
+  if (is_hms_module_mandatory_)
+  {
+    if (!init_hms())
+    {
+      rclcpp::shutdown();
+      return CallbackReturn::FAILURE;
+    }
+  }
+
```

Comment:
This is a bit tricky. The reason why it is not bundled up with the rest of modules is because the callback on the HMS messages uses a ROS2 publisher, which needs to be instantiated by the time said callback is registered. I did not find a way to do that without affecting the current code structure, do you have any ideas here?

Reply:
It's true. Now I remember that you mentioned it to me. We can leave it like this, but the functionality should be a bit different, to at least match the other modules. So, in case it is not mandatory, it will still try to initialize it but if it fails, the node will continue without reporting any error.